app/lib/actions/katello/repository/update.rb

```diff
@@ -28,8 +28,13 @@ def plan(root, repo_params)
                                        label: content.label,
                                        gpg_url: repository.yum_gpg_key_url)
           end
+          if root.pulp_update_needed?
+            plan_action(PulpSelector,
+                        [::Actions::Pulp::Repository::Refresh,
+                         Pulp3::Orchestration::Repository::Update],
+                        repository, SmartProxy.pulp_master)
+          end
 
-          plan_action(::Actions::Pulp::Repository::Refresh, repository) if root.pulp_update_needed?
           plan_self(:repository_id => root.library_instance.id)
         end
 
```

Comment:
Should we trigger a metadata generate on updates in the run phase where pulp update is not needed? Doesn't feel necessay?

Reply:
Good question... can we just put it in the same if clause when checking if an update is needed?

Reply:
I think this change would be safe to make.